### DISCUSSION TITLE
Prisma & ESM: works in dev, breaks in build

### DISCUSSION THREAD

**Discussion by zigomir** [nuxt/nuxt#17855] [Questions] [answered]
Hi

Not sure if this is Nuxt, Prisma, Vite or Rollup issue 😊 (I suspect Prisma), but I'm getting this error on `npm run build`

```
ERROR  Rollup error: ENOENT: no such file or directory, copyfile '/Users/ziga/development/nuxt3-app/node_modules/.prisma/package.json' -> '/Users/ziga/development/nuxt3-app/.output/server/node_modules/.prisma/package.json'
```

I'm using Prisma in `server/api/count.ts` like so

```ts
// import { PrismaClient } from '@prisma/client' // won't work
// https://github.com/prisma/prisma/issues/6491
// https://github.com/prisma/prisma/issues/9142
import PrismaClientPkg from '@prisma/client'
const PrismaClient = PrismaClientPkg.PrismaClient

const prisma = new PrismaClient()

let counter = 0
export default async () => {
  counter++

  const allUsers = await prisma.user.findMany()
  console.log(allUsers)

  return {
    counter,
  }
}
```

Before submitting an issue I wonder where do you think this issue lies.
Thanks!

[the remark this answers is not in the document]
**Reply by lihbr** [reply]
I work @prismicio 🙃 not @prisma haha

Anyway, not really sure what's going on here unfortunately ^^' I guess it's Prisma package relying on the `package.json`? Should get embedded though 🤔

**Comment by tobiasdiez** [comment]
Just adding that 
```
import { xyz } from '@prisma/client';
```
is not possible because `prisma/client` is a CommonJS module and ES6 compatibility is still an open feature request https://github.com/prisma/prisma/issues/5030.

**Reply by zigomir** [reply]
Right, that's true. So maybe I wonder more how come that in dev it works but the build fails.

I guess I should isolate the scope and prepare repro with `vite` only first 🤔

[the remark this answers is not in the document]
**Reply by pi0** [reply]
I cannot believe read it wrongly and mentioning you. Sorry 🙈  Must get some sleep.

**Comment by zigomir** [comment, marked as answer]
Ok I got something that works for me, I just need to manually copy a non existing file to right place

```diff
- "build": "nuxt build",
+ "build": "cp node_modules/.prisma/client/package.json node_modules/.prisma && nuxt build",
```

**Comment by tobiasdiez** [comment]
@zigomir did you find a nice workaround for the esm shortcomings of prisma? Of course, I could replace all `import {...} from `@prisma/client` constructions with top-level imports and then destructure it later; but this feels kind of inconvenient.

**Reply by zigomir** [reply]
No, I haven't :/ I guess issue is on "[their side](https://github.com/prisma/prisma/issues/5030#issuecomment-912588323)".

**Comment by lewebsimple** [comment]
I have documented an approach to use Prisma with Nuxt3 in this article : https://dev.to/lewebsimple/declarative-database-modelling-2h8d

**Reply by zigomir** [reply]
Thanks for the note on `Prisma Data Proxy`. I was getting `$fetch is not defined` error and was completely in the dark about why this error is happening.

**Comment by husayt** [comment]
As prisma still doesn't support ESM builds, what is the best way to use commonjs library from nuxt, especially if it is not in node_modules. Prisma client is a generated code (like `.nuxt`) and I would rather keep it outside node_modules, but same time it is commonjs and I need nuxt to handle it accordingly
